src/main/java/emu/grasscutter/server/packet/recv/HandlerHitTreeNotify.java

```diff
@@ -0,0 +1,69 @@
+package emu.grasscutter.server.packet.recv;
+
+import java.util.HashMap;
+
+import emu.grasscutter.data.GameData;
+import emu.grasscutter.game.entity.EntityItem;
+import emu.grasscutter.game.player.Player;
+import emu.grasscutter.game.world.Scene;
+import emu.grasscutter.net.packet.Opcodes;
+import emu.grasscutter.net.packet.PacketHandler;
+import emu.grasscutter.net.packet.PacketOpcodes;
+import emu.grasscutter.net.proto.VectorOuterClass;
+import emu.grasscutter.server.game.GameSession;
+import emu.grasscutter.net.proto.HitCollisionOuterClass.HitCollision;
+import emu.grasscutter.utils.AutoRecycleHashMap;
+import emu.grasscutter.utils.Position;
+
+@Opcodes(PacketOpcodes.HitTreeNotify)
+public class HandlerHitTreeNotify extends PacketHandler {
+    private final static int RECORD_EXPIRED_SECONDS = 60*5;
+    private final static AutoRecycleHashMap<Integer,HashMap<Integer,HitTreeRecord>> hitRecord = new AutoRecycleHashMap<>(RECORD_EXPIRED_SECONDS);

```

Comment:
Why don't you just move this to `Player`, as Melledy had suggested? This `AutoRecycleHashMap` just adds a lot of unnecessary complexity.

Reply:
> Why don't you just move this to `Player`, as Melledy had suggested? This `AutoRecycleHashMap` just adds a lot of unnecessary complexity.

all done